recipes/poly2tri/all/CMakeLists.txt

```diff
@@ -11,6 +11,11 @@ add_library(poly2tri
     ${POLY2TRI_SRC_DIR}/sweep/sweep_context.cc
 )
 
+# _USE_MATH_DEFINES is defined in utils.h where M_PI is used, but it can break because math.h or cmath
+# may have already been included indirectly by other headers included by sweep.cc before inclusion of utils.h
+# Therefore we ensure that _USE_MATH_DEFINES is defined before any inclusion of headers.
+target_compile_definitions(poly2tri PRIVATE _USE_MATH_DEFINES)
```

Comment:
Question, any reason to prefer this against adding `tc.defines["..` in the `generate()` method?

Reply:
For CMake projects, using the `target_*` scoped functions is generally considered best practice. I think this also fits better as part of the build system where possible.

Reply:
Yep, we have a custom CMakeLists, there is no reason to inject definitions through tc.preprocessor_definition rather than CMakeLists itself.